test/test_maybe.cxx

```diff
@@ -517,3 +517,94 @@ TEST(Maybe, reset) {
   mbPinned.reset();
   ASSERT_TRUE(mbPinned.isNothing());
 }
+
+TEST(Maybe, is_maybe) {
+  ma::Maybe<int> a{1};
+  ASSERT_TRUE(ma::is_maybe_v<decltype(a)>);
+  ASSERT_TRUE(ma::is_maybe_v<decltype(std::move(a))>);
+  ASSERT_FALSE(ma::is_maybe_v<int>);
+}
+
+TEST(Maybe, mapN) {
+  // We first create an object that is only movable, and need if for several
+  // reasons:
+  //  - To get compiler errors if a move is not handled correctly
+  //  - To check if an lvalue reference was moved from
+  class Movable {
+   public:
+    explicit Movable(std::string value_) : value(value_) {}
+    Movable(Movable&& other) : value(std::move(other.value)) {}
+    Movable& operator=(Movable&& other) {
+      value = std::move(other.value);
+      return *this;
+    }
+    Movable(const Movable&& other) = delete;
+    Movable& operator=(const Movable& other) = delete;
+
+    size_t size() const { return value.size(); }
+
+   private:
+    std::string value;
+  };
```

Comment:
nit: there's an existing `NoCopy_t` defined earlier where you can also verify that it was moved from